### PR TITLE
Format and fix clippy warnings for new Rust stable version

### PR DESCRIPTION
**Pull request by shaobo-he-aws**
## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [x] A change "invisible" to users (e.g., documentation, changes to "internal" crates like `cedar-policy-core`, `cedar-validator`, etc.)

I confirm that this PR (choose one, and delete the other options):

- [x] Does not update the CHANGELOG because my change does not significantly impact released code.

I confirm that [`cedar-spec`](https://github.com/cedar-policy/cedar-spec) (choose one, and delete the other options):

- [x] Does not require updates because my change does not impact the Cedar Dafny model or DRT infrastructure.

## Disclaimer

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by john-h-kastner-aws — APPROVED** [review]

File: cedar-policy-formatter/src/pprint/doc.rs [hunk 1]
```diff
@@ -280,24 +280,21 @@ impl Doc for ASTNode<Option<Add>> {
...
-                                    RcDoc::nil(),
-                                ))
-                                .append(RcDoc::line())
-                                .append(e.to_doc(context)),
-                            e,
-                        ))
-                    },
-                )?
+                .try_fold((initial.to_doc(context)?, initial), |pair, (op, e)| {
+                    let pair = pair;
```
Comment (cdisselkoen) [inline cedar-policy-formatter/src/pprint/doc.rs:284]: is this necessary?

File: cedar-policy-formatter/src/pprint/doc.rs [hunk 2]
```diff
@@ -320,24 +317,21 @@ impl Doc for ASTNode<Option<Mult>> {
...
```
Comment (cdisselkoen) [inline cedar-policy-formatter/src/pprint/doc.rs:321]: is this necessary?

**Review by cdisselkoen — APPROVED** [review]

**Review by aaronjeline — APPROVED** [review]

File: cedar-policy-validator/src/types.rs [hunk 3]
```diff
@@ -325,8 +325,8 @@ impl Type {
         schema: &ValidatorSchema,
         tys: &[Type],
     ) -> Option<Type> {
-        tys.iter().fold(Some(Type::Never), |lub, next| {
```
[the remark this answers is not in the document]
Comment (john-h-kastner-aws) [reply cedar-policy-validator/src/types.rs:328]: I'm convinced it's the same

File: cedar-policy-formatter/src/pprint/doc.rs (hunk 2 above)
Comment (shaobo-he-aws) [reply cedar-policy-formatter/src/pprint/doc.rs]: Good catch. Fixed.

File: Cargo.toml [hunk 4]
```diff
@@ -7,9 +7,8 @@ members = [
 	"cedar-policy-cli",
 ]
 
+resolver = "2"
```
[the remark this answers is not in the document]
Comment (shaobo-he-aws) [reply Cargo.toml:10]: I think we don't specify edition number in the top-level Cargo.toml file, which makes the resolver version to be 1. Other Cargo.toml files do and have resolver versions to be 2.

**Review by john-h-kastner-aws — APPROVED** [review]

**Pull request merged**